.github/workflows/main.yml

```diff
@@ -11,18 +11,22 @@ jobs:
   # This workflow contains a single job called "build"
   build:
     # The type of runner that the job will run on
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
 
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
     # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
     - uses: actions/checkout@v2
 
-    # Install dependencies.
+    # Install dependencies
     - name: Install dependencies
-      run: sudo apt install cmake g++ libjpeg-dev libpng-dev libprotobuf-dev libwebp-dev ninja-build protobuf-compiler
+      run: sudo apt install cmake g++ libjpeg-dev libpng-dev libprotobuf-dev libwebp-dev libgtest-dev ninja-build protobuf-compiler valgrind
 
     # Build the code
     - name: Build
       run: |
-        mkdir build && cd build && cmake -G "Ninja" ../ && ninja
+        mkdir build && cd build && cmake -G "Ninja" ../ && ninja 
+    
+    # Run tests
+    - name: Run tests
+      run: valgrind --leak-check=full --show-leak-kinds=all ./build/test/test_thumbnailer
```

Comment:
So you are right that there are a bunch of memory leaks! Without any options, valgrind only shows a summary of leaks with the number of blocks lost (definitely lost, indirectly lost...). This is already an error! If there are no leaks it shows "All heap blocks were freed -- no leaks are possible" instead. The flags allow showing extra info to help debugging.

I found two memory leaks in the test file, but there are also leaks in the Thumbnailer class. Those can be fixed in another commit.
- In class.cc on line 56, a new encoder is created every time GenerateAnimationNoBudget is called, but only the last one gets deleted in the destructor. Either we should make sure to only create one encoder, or the previous one must be deleted before creating a new one.
- There are also issues with the life cycle of webp_data/new_webp_data. On line 132, we are only copying the data pointer, not the actual data, so it probably gets deleted later on line 103? and I think there are WebPDataClear calls missing for webp_data and/or webp_new_data. I haven't looked into this fully so it would be great if one of you two could look into fixing it properly.